### ISSUE TITLE
Example in README does not work with the default stylesheet

### ISSUE THREAD

**Issue by AirQuick** [transpect/css-tools#2]
Using the example in [README](https://github.com/transpect/css-tools/blob/e77272a7b814870a12f6e31f54a176af79a12ff0/README.md), I was able to get the expected output:

```xml
    <p class="red" css:color="red">This text has the color red.</p>
```

However, with the same example, using the default stylesheet (i.e. removing this element

```xml
    <p:input port="stylesheet">
      <p:document href="../xsl/css-parser.xsl"/>
    </p:input>
```

from `css:expand`), it didn't work:

```xml
    <p class="red">This text has the color red.</p>
```

**Comment by gimsieke** [comment]
We are currently migrating from a regex-based parser to a grammar-based parser. I think I already fixed #1 with 9916cd0 and I will look into this issue later today.

**Comment by AirQuick** [comment]
It might be possibly that applying `post-process` is missing from [a template](https://github.com/transpect/css-tools/blob/master/xsl/REx_css-parser.xsl#L94).

**Comment by gimsieke** [comment]
This looks very plausible, thanks for investigating.
We’ve been busy improving the grammar so that comments be ignored more reliably (we committed that already in 9e27daa and e6e1a0f) and currently I’m still refactoring the processing so that we have distinct XSLT invocations and we can see/debug the parser output of styles included by `link`, `style`, and `@import`, prior to postprocessing. 
I’m confident that we will fix your issue shortly, too. I was thinking about encouraging you to file a pull request for this particular template, but I’m afraid that there might be too many changes that could also affect this template. So please hold on for another couple of hours…

**Comment by gimsieke** [comment]
It will take a bit longer. We are having severe issues with whitespace/comments. The root cause of which is that whitespace may be a combinator in selectors. We tried to treat comments as whitespace, but then the [parser generator](http://bottlecaps.de/rex/) won’t let us use comments in selectors due to lexical ambiguities. We could avoid this error by ignoring whitespace also in selectors, but then we were not able to discern `.foo .bar` from `.foo.bar`. We are inclined to strip comments away completely before the parsing. But then we remembered that some EPUB customers want to be able to retain the comments. In the previous regex-based implementation, we pulled out comments in rules (including selectors) and put them before or after the rule. But this is not possible with a grammar-based parser. A possible way out might be that by default comments are stripped away before the parsing, but there is an option to retain them. But then the grammar would only permit them between rules and fail to parse the CSS if they are anywhere else. I will discuss this with @fr4nze today.
